src/main/java/com/example/usermanagement/controller/AuthController.java

```diff
@@ -25,9 +33,25 @@ public ResponseEntity<ResponseDto> register(@Valid @RequestBody CreateUserDto cr
         return new ResponseEntity<>(userService.create(createUserdto), HttpStatus.CREATED);
     }
 
-    @PostMapping("authenticate")
-    public ResponseEntity<AuthTokenResponseDto> authenticate(@Valid
-            @RequestBody LoginDto loginDto) {
-        return new ResponseEntity<>(userService.authenticate(loginDto), HttpStatus.OK);
+    @PostMapping("login")
+    public ResponseEntity<String> authenticate(@Valid
+                                                   @RequestBody LoginDto loginDto) {
+        String tokenResponse = userService.authenticate(loginDto);
+        if (tokenResponse != null) {
+            return new ResponseEntity<>(tokenResponse, HttpStatus.OK);
+        } else {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
```

Comment:
should it be unauthorized or bad request?

Reply:
If the credentials or token provided are wrong than the user is not authenticated to access the api, so its unauthorized.